src/quo/components/wallet/token_input/view.cljs

```diff
@@ -7,85 +7,121 @@
     [quo.components.tags.network-tags.view :as network-tag]
     [quo.components.utilities.token.view :as token]
     [quo.components.wallet.token-input.style :as style]
-    [quo.foundations.colors :as colors]
     [quo.foundations.common :as common]
     [quo.theme :as quo.theme]
     [react-native.core :as rn]
     [reagent.core :as reagent]))
 
 (defn calc-value
   [crypto? currency token value conversion]
-  (let [num-value (if (string? value) (parse-double (or value "0")) value)]
+  (let [num-value (if (string? value)
+                    (or (parse-double value) 0)
+                    value)]
     (if crypto?
       (str (get common/currency-label currency) (.toFixed (* num-value conversion) 2))
-      (str (.toFixed (/ num-value conversion) 2) " " (string/upper-case (or (clj->js token) ""))))))
+      (str (.toFixed (/ num-value conversion) 2)
+           " "
+           (string/upper-case (or (clj->js token) ""))))))
```

Comment:
We need a function that takes token symbols (`:eth`) and return some user friendly string ("ETH"), I thought it already existed, but I couldn't find it.

These conversions are very error-prone, and they've given us some runtime crashes

Reply:
Good point @ulisesmac. I have noticed that too in status-mobile. String manipulation in Clojure doesn't follow nil punning like collection functions do. It's a common source of error. I think the error prone aspect also comes from the lack of this knowledge. 

I believe it's one of the reasons we have functions like `safe-trim` or `safe-replace`, because some mobile developers got tired of these nil pointer exceptions and the need to remember to check for nils, so they decided to wrap core cljs string functions. I did this too.

But these wrapping functions in `utils.string` are a fragile solution and require this effort to wrap cljs functions and make the code more verbose (longer function names).

To me, all of us should just remember that almost anything can be nil in mobile, therefore, string manipulation should be considered unsafe without `some->` or `or`.

Reply:
Yes, sounds good, but, is there a way to override the string implementation?

TBH, I see this as a ClojureScript error, they are not consistent with their implementation, see `string/blank?`:
![image](https://github.com/status-im/status-mobile/assets/90291778/6a0ad980-7f0a-403e-a040-47868f5c07fa)

They are calling `gstring/makeSafe` why not call it in other places too?

I think it'd be ok if we just override some cljs.core implementations to add a call to `makeSafe`. I know, probably not the best, but IMO they are also not doing the best.